be/src/vec/core/field.h

```diff
@@ -572,13 +671,21 @@ class Field {
         create(reinterpret_cast<const char*>(data), size);
     }
 
+    void create_json(const unsigned char* data, size_t size) {
+        new (&storage) JsonField(reinterpret_cast<const char*>(data), size);
+        which = Types::JSON;
+    }
+
     ALWAYS_INLINE void destroy() {
         if (which < Types::MIN_NON_POD) return;
 
         switch (which) {
         case Types::String:
             destroy<String>();
             break;
+        case Types::JSON:
+            destroy<JsonField>();
```

Comment:
where is the implementation of destroy<JsonField>() ?

Reply:
```C++
    template <typename T>
    void destroy() {
        T* MAY_ALIAS ptr = reinterpret_cast<T*>(&storage);
        ptr->~T();
    }
```
[Line707](https://github.com/freesinger/incubator-doris/blob/878d392fb89991c9e80f9fff0936fe3ce97350de/be/src/vec/core/field.h#L707)